File: templates/select_miss_het_qcplink.pl
#!/usr/bin/env perl 

 
                      

\$cut_het_high="${params.cut_het_high}";
\$cut_het_low="${params.cut_het_low}";
\$cut_miss="${params.cut_miss}";

open(MISSFILE,"$imiss");
open(HETFILE,"$het");
@all=<HETFILE>;
chomp(@all);
open(OUT,">$outfname");

\$line=0;
while(\$data = <MISSFILE>){
chomp(\$data);
  if(\$line>=1){
    chomp(\$data);
    @parts_miss=split(/\\s+/,\$data);
    \$missing=\$parts_miss[6];
    @parts_het=split(/\\s+/,\$all[\$line]);
    \$meanHet=sprintf("%.3f", (\$parts_het[5]-\$parts_het[3])/\$parts_het[5]);

    if(\$missing>\$cut_miss or \$meanHet>\$cut_het_high or \$meanHet<\$cut_het_low){
print OUT \$parts_miss[1],"\t",\$parts_miss[2],"\t",\$missing,"\t",\$meanHet,"\n";
}
}


++\$line;
}

File: templates/select_miss_het_qcplink.py
#!/usr/bin/env python

# This script selects those who fail heterozygosity constraints -- previous versions also looked at missingness too
# hence the name


from pandas import read_csv
import sys

TAB = chr(9)
if len(sys.argv)<=1:
  sys.argv = ["select_miss_het.qcplink.py","$het",float("${params.cut_het_low}"),float("${params.cut_het_high}"),"$outfname"]

hetf = sys.argv[1]
cut_het_high=float(sys.argv[3])
cut_het_low=float(sys.argv[2]);
outfname = sys.argv[4]


het      = read_csv(hetf,delim_whitespace=True)
mean_het = (het["N(NM)"]-het["O(HOM)"])/het["N(NM)"]
failed   = het[(mean_het<cut_het_low) | (mean_het>cut_het_high)]

failed.to_csv(outfname,header=False,sep=TAB)
